### ISSUE TITLE
Chinese pinyin support 

### ISSUE THREAD

**Issue by absir1949** [8bitPit/Niagara-Issues#234]
just like microsoft lanuacher.

it will combine Letters and Pinyin.

**Comment by 8bitPit** [comment]
I would, but unfortunately I don't know how don't know how that system works, but I will do some research, If anyone can help, please let me know.

**Comment by OldFanta** [comment]
@8bitPit I think it is a good idea to convert Chinese characters into pinyin first. Pinyin is like an English letter, such as "百度" = "Baidu".
Or you can refer to  [Project(Pinyin)](https://github.com/hotoo/pinyin), [Project(PinyinOrder)](https://github.com/imwangji/pinyinOrder), [Article(Android PinyinOrder)](https://blog.csdn.net/holemole/article/details/6640934) and [Article(Android SQL Chinese Order)](https://blog.csdn.net/androidxiaogang/article/details/66974142)(sorry I have not studied programming, but those looks reliable.)
Anyway, the concept of this launcher is great, thank you for your wonderful job, and look forward to the emergence of Chinese sorting!

**Comment by fanjl** [comment]
Chinese user need this

**Comment by xrkxrk** [comment]
http://pinyin4j.sourceforge.net/
I think this may help you. @8bitPit

**Comment by 8bitPit** [comment]
@xrkxrk sadly, I cannot use your suggested library, since it's licensed under the copyleft GNU license.

**Comment by 8bitPit** [comment]
@OldFanta thanks for your help. However, Pinyin libraries hafe to be written in Java / Kotlin, so they can be used for Android apps.

**Comment by 8bitPit** [comment]
I could use: https://github.com/promeG/TinyPinyin or https://github.com/KyleTang/JPinYin.

**Comment by OldFanta** [comment]
> I could use: https://github.com/promeG/TinyPinyin or https://github.com/KyleTang/JPinYin.

I am very glad to hear this news, thank you very much. Looking forward to seeing in the follow-up update!

**Comment by xrkxrk** [comment]
Thanks a million, Look forward to your success!

**Comment by xrkxrk** [comment]
> I could use: https://github.com/promeG/TinyPinyin or https://github.com/KyleTang/JPinYin.

Thanks a million, Look forward to your success!

**Comment by Scofieldsu** [comment]
look forward to your success，and i also expect that searching app can fuzzy  query  by Chinese Pinyin。

like it very much @8bitPit  （づ￣3￣）づ╭❤～

**Comment by 8bitPit** [comment]
@xinkunZ also suggested a library to support pinyin: https://sourceforge.net/projects/pinyin4j/

**Comment by xinkunZ** [comment]
@8bitPit   If you choose pinyin4j to translate app name to pinyin,  there is an example
```
  private String getPinyin(String appName, String defaultName) {
    try {
      HanyuPinyinOutputFormat format = new HanyuPinyinOutputFormat();
      format.setCaseType(HanyuPinyinCaseType.LOWERCASE);
      format.setToneType(HanyuPinyinToneType.WITHOUT_TONE);
      StringBuilder sb = new StringBuilder();
      char[] chars = appName.toCharArray();
      for (char aChar : chars) {
        String[] strings = PinyinHelper.toHanyuPinyinStringArray(aChar, format);
        if (strings != null && strings.length > 0) {
          sb.append(strings[0]);
        } else {
          sb.append(aChar);
        }
      }
      return sb.toString();
    } catch (Exception e) {
      return defaultName;
    }
  }
```
* `System.out.println(getPinyin("中国", "中国"));`，then you can get 'zhongguo' in console
* And you can check Chinese pinyin use google translate
![image](https://user-images.githubusercontent.com/7759897/52703645-c9a50080-2fb9-11e9-91fa-279a503b5d12.png)

* Also, if you add app search in home page, I hope you can support use pinyin to search apps

**Really like this launcher!! Thanks!**

**Comment by 8bitPit** [comment]
Thanks for helping @xinkunZ :)

**Comment by Maxr1998** [comment]
Honestly, TinyPinyin seems to be the best option for me. Especially since it has low memory requirements and is said to be a lot faster. @8bitPit

**Comment by 2236125280** [comment]
Maybe you can try this
https://www.cnblogs.com/goloving/p/7662676.html

**Comment by 8bitPit** [comment]
I've discovered Android has an inbuilt Pinyin transliterator. It is somewhat hidden inside the framework, but I can use it on Android 7 and above. So no library is need, thanks for all your suggestions though.

**Comment by 8bitPit** [comment]
Implemented in v0.9.9: https://github.com/8bitPit/Niagara-Issues/releases
